src/org/traccar/protocol/Arnavi4ProtocolDecoder.java

```diff
@@ -0,0 +1,187 @@
+import static org.traccar.protocol.Arnavi4FrameDecoder.PACKAGE_END_SIGN;
+
+public class Arnavi4ProtocolDecoder extends BaseProtocolDecoder {
+
+    private static final byte RECORD_PING = 0x00;
+    private static final byte RECORD_DATA = 0x01;
+    private static final byte RECORD_TEXT = 0x03;
+    private static final byte RECORD_FILE = 0x04;
+    private static final byte RECORD_BINARY = 0x06;
+
+    private static final byte TAG_LATITUDE = 3;
+    private static final byte TAG_LONGITUDE = 4;
+    private static final byte TAG_COORD_PARAMS = 5;
+
+    public Arnavi4ProtocolDecoder(Arnavi4Protocol protocol) {
+        super(protocol);
+    }
+
+    private Position decodePosition(DeviceSession deviceSession, ChannelBuffer buf, int length, Date time) {
+
+        final Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        position.setTime(time);
+
+        int readBytes = 0;
+        while (readBytes < length) {
+            short tag = buf.readUnsignedByte();
+            switch (tag) {
+                case TAG_LATITUDE:
+                    position.setLatitude(buf.readFloat());
+                    position.setValid(true);
+                    break;
+
+                case TAG_LONGITUDE:
+                    position.setLongitude(buf.readFloat());
+                    position.setValid(true);
+                    break;
+
+                case TAG_COORD_PARAMS:
+                    position.setCourse(buf.readUnsignedByte() * 2.0);
+                    position.setAltitude(buf.readUnsignedByte() * 10.0);
+                    byte satellites = buf.readByte();
+                    position.set(Position.KEY_SATELLITES, satellites & 0x0F + (satellites >> 4) & 0x0F); // gps+glonass
+                    position.setSpeed(buf.readUnsignedByte() * 1.852);
+                    break;
+
+                default:
+                    buf.readBytes(4); // Skip other tags
+                    break;
+            }
+
+            readBytes += 5; // 1 byte tag + 4 bytes value
+        }
+
+        return position;
+    }
+
+    @Override
+    protected Object decode(
+            Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        ChannelBuffer buf = (ChannelBuffer) msg;
+
+        byte startSign = buf.readByte();
+
+        if (startSign == HEADER_START_SIGN) {
+
+            byte version = buf.readByte();
+
+            String imei = String.valueOf(buf.readLong());
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, imei);
+
+            if (deviceSession != null && channel != null) {
+
+                final ChannelBuffer response;
+
+                if (version == HEADER_VERSION_1) {
+                    response = ChannelBuffers.dynamicBuffer(ByteOrder.LITTLE_ENDIAN, 4);
+                    response.writeBytes(new byte[]{0x7B, 0x00, 0x00, 0x7D});
```

Comment:
I think we should create a separate method to send response. Please check GT06 or some other binary protocol for an example.